plutus-tx/src/PlutusTx/List.hs

```diff
@@ -170,10 +170,9 @@ splitAt n xs
   where
     go :: Integer -> [a] -> ([a], [a])
     go _ []     = ([], [])
-    go 1 (y:ys) = ([y], ys)
```

Comment:
It's actually interesting because we also use `| n <= 0` in a few functions, and that's fine, but matching on number literals isn't. Any idea why? Is `Eq` more likely to be specialized than `Ord`?